### PR TITLE
Enable Password Complexity Compliance

### PR DESCRIPTION
**Pull request by edtubillara**
This commit enables the Keystone security compliance
feature for password complexity, strengthens the default passwords,
removes the jellyroll password validator from the pipelines, and adds the
horizon validator.

### CODE REVIEW

File: roles/horizon/defaults/main.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@ horizon:
   session_engine: django.contrib.sessions.backends.cache
   osp_sys_packages:
     - neutron-lbaas-dashboard
+  password_validator:
+    enabled: True
+    regex: '^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&+=()_]).{8,}$'
+    help_text: 'Password is not compliant. Password must contain uppercase, lowercase, special characters !@#$%%^&+=()_, digits, and be greater than 8 characters in length.'
```
Comment (lihkin213) [inline roles/horizon/defaults/main.yml]: It is >= 8 or >8 chars

File: roles/keystone/templates/etc/keystone/keystone-paste.ini [hunk 2]
```diff
@@ -78,13 +75,13 @@ use = egg:keystone#service_v3
 use = egg:keystone#admin_service
 
 [pipeline:public_api]
-pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension{{ ' passval unauthlog' if keystone.jellyroll|bool else '' }} public_service
+pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension{{ ' unauthlog' if keystone.jellyroll|bool else '' }} public_service
 
 [pipeline:admin_api]
-pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension s3_extension{{ ' passval unauthlog' if keystone.jellyroll|bool else '' }} admin_service
+pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension{{ ' unauthlog' if keystone.jellyroll|bool else '' }} s3_extension admin_service
```
Comment (lihkin213) [inline roles/keystone/templates/etc/keystone/keystone-paste.ini]: is there a reason why the s3_extension middleware was moved after jellyroll?

File: roles/keystone/templates/etc/keystone/keystone-paste.ini [hunk 3]
```diff
@@ -78,13 +75,13 @@ use = egg:keystone#service_v3
...
 
 [pipeline:api_v3]
-pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension_v3 s3_extension{{ ' rbac_filter passval unauthlog' if keystone.jellyroll|bool else '' }} service_v3
+pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension_v3{{ ' rbac_filter unauthlog' if keystone.jellyroll|bool else '' }} s3_extension service_v3
```
Comment (lihkin213) [inline roles/keystone/templates/etc/keystone/keystone-paste.ini]: Why is this moved after the jellyroll

File: roles/keystone/templates/etc/keystone/keystone-paste.ini [hunk 4]
```diff
@@ -78,13 +75,13 @@ use = egg:keystone#service_v3
 use = egg:keystone#admin_service
 
 [pipeline:public_api]
-pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension{{ ' passval unauthlog' if keystone.jellyroll|bool else '' }} public_service
+pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension{{ ' unauthlog' if keystone.jellyroll|bool else '' }} public_service
 
 [pipeline:admin_api]
-pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension s3_extension{{ ' passval unauthlog' if keystone.jellyroll|bool else '' }} admin_service
+pipeline = healthcheck cors sizelimit url_normalize http_proxy_to_wsgi request_id build_auth_context token_auth json_body ec2_extension{{ ' unauthlog' if keystone.jellyroll|bool else '' }} s3_extension admin_service
```
Comment (lihkin213) [inline roles/keystone/templates/etc/keystone/keystone-paste.ini]: what is the s_extension moved after jellyroll

File: roles/keystone-defaults/defaults/main.yml [hunk 5]
```diff
@@ -40,6 +40,10 @@ keystone:
     - keystone-wsgi-admin
     - keystone-wsgi-public
   jellyroll: False # custom middleware for password compliance
+  security_compliance:
+    enabled: true
+    password_regex: '^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%^&+=()_]).{8,}$'
+    password_regex_description: 'Password is not compliant. Password must contain uppercase, lowercase, special characters !@#$%%^&+=()_, digits, and be greater than 8 characters in length.'
```
Comment (lihkin213) [inline roles/keystone-defaults/defaults/main.yml]: >8 chars. This should be reworded to 8 or more chars.

File: roles/keystone/templates/etc/keystone/keystone-paste.ini (hunk 4 above)
Comment (edtubillara) [inline roles/keystone/templates/etc/keystone/keystone-paste.ini]: good catch

**Comment by lihkin213** [comment]
ok to test

**Comment by nirajdp76** [comment]
ok to test
